FreeRTOS/iot-reference-stm32u5
# Handled persistent session mutex relinquish correctly

Description
-----------
This PR makes changes required to release a mutex for subscription manager context held by the MQTT agent task when there is a persistent session.
Without this change, there is a potential for deadlock as explained in this Issue https://github.com/FreeRTOS/iot-reference-stm32u5/issues/81 by @grdSTM and @gmarcolinosr.

Test Steps
-----------


Checklist:
----------


- [ NA ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
https://github.com/FreeRTOS/iot-reference-stm32u5/issues/81.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
